src/librustc/ty/layout.rs

```diff
@@ -2504,7 +2504,7 @@ where
         };
 
         let mut inputs = sig.inputs();
-        let extra_args = if sig.abi == RustCall {
+        let extra_args = if sig.abi == RustCall && !sig.inputs().is_empty() {
```

Comment:
This is just ignoring the error. We should be hitting the `bug!` below if we don't have any arguments. You just need to make the match more resilient.

Reply:
So, we should emit ICE like [this](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=ace01930647bb537a666c48f9eef33ec) instead of current one?

Reply:
Yes: the ICE with the error message is more informative and due to `unboxed_closures` being an unfinished feature. (So there's no need to add a test for this feature, as it's expected to ICE.)

Reply:
Got it, fixed and removed test. Shouldn't we close that issue via this PR?